compiler/rustc_codegen_rmc/src/context/goto_ctx.rs

```diff
@@ -56,14 +55,13 @@ pub struct GotocCtx<'tcx> {
 /// Constructor
 impl<'tcx> GotocCtx<'tcx> {
     pub fn new(tcx: TyCtxt<'tcx>) -> GotocCtx<'tcx> {
-        let (thks, fhks) = type_and_fn_hooks();
+        let fhks = fn_hooks();
```

Comment:
You're not just removing those hooks, but the idea of type hooks entirely.  Is there any case where we might want one?

Reply:
I was hoping to get rid of all hooks eventually. :)

I don't think we should keep dead code here. It's fairly small and easy to rewrite if we ever need it again.